Common/TableProducer/PID/pidTPC.cxx

```diff
@@ -110,102 +112,66 @@ struct tpcPid {
-
-      ccdbPathSigma = ccdbPath.value + "/" + sigmaname.value;
-      LOG(info) << "Loading exp. sigma parametrization from CCDB, using path: " << ccdbPathSigma << " for timestamp " << ccdbTimestamp.value;
-      response.LoadParam(DetectorResponse::kSigma, ccdb->getForTimeStamp<Parametrization>(ccdbPathSigma, ccdbTimestamp.value));
+      const std::string path = ccdbPath.value;
+      const auto time = ccdbTimestamp.value;
+      ccdb->setURL(url.value);
+      ccdb->setTimestamp(time);
+      ccdb->setCaching(true);
+      ccdb->setLocalObjectValidityChecking();
+      ccdb->setCreatedNotAfter(std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count());
+      response.SetParameters(ccdb->getForTimeStamp<o2::pid::tpc::Response>(path, time));
+      LOGP(info, "Loading TPC response from CCDB, using path: {} for ccdbTimestamp {}", path, time);
+      response.PrintAll();
     }
   }
 
-  template <o2::track::PID::ID pid>
-  using ResponseImplementation = o2::pid::tpc::ELoss<Trks::iterator, pid>;
-  void process(Trks const& tracks,
-               aod::BCsWithTimestamps const&,
-               Coll const&)
+  void process(Coll const& collisions, Trks const& tracks,
+               aod::BCsWithTimestamps const&)
   {
-    constexpr auto responseEl = ResponseImplementation<PID::Electron>();
-    constexpr auto responseMu = ResponseImplementation<PID::Muon>();
-    constexpr auto responsePi = ResponseImplementation<PID::Pion>();
-    constexpr auto responseKa = ResponseImplementation<PID::Kaon>();
-    constexpr auto responsePr = ResponseImplementation<PID::Proton>();
-    constexpr auto responseDe = ResponseImplementation<PID::Deuteron>();
-    constexpr auto responseTr = ResponseImplementation<PID::Triton>();
-    constexpr auto responseHe = ResponseImplementation<PID::Helium3>();
-    constexpr auto responseAl = ResponseImplementation<PID::Alpha>();
-
-    auto reserveTable = [&tracks](const Configurable<int>& flag, auto& table) {
-      if (flag.value != 1) {
-        return;
+    // Check and fill enabled tables
+    auto makeTable = [&tracks, &collisions, this](const Configurable<int>& flag, auto& table, const o2::track::PID::ID pid) {
+      if (flag.value == 1) {
+        // Prepare memory for enabled tables
+        table.reserve(tracks.size());
+        int lastCollisionId = -1;        // Last collision ID analysed
+        for (auto const& trk : tracks) { // Loop on Tracks
+          auto collision = collisions.iteratorAt(trk.collisionId());
+          if (useCCDBParam && ccdbTimestamp.value == 0 && trk.has_collision() && trk.collisionId() != lastCollisionId) { // Updating parametrization only if the initial timestamp is 0
+            lastCollisionId = trk.collisionId();
+            const auto& bc = collision.bc_as<aod::BCsWithTimestamps>();
+            response.SetParameters(ccdb->getForTimeStamp<o2::pid::tpc::Response>(ccdbPath.value, bc.timestamp()));
```

Comment:
Unless I misunderstood doing so checks for the ccdb up to 9 times for all tracks, this might be a bit expensive and is probably not needed as in the previous implementation

Reply:
I changed this, so that this part is done before the make Table where it would be called 9 times.